common.sh

```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env bash
+
+HAS_HELM="$(type "helm" &> /dev/null && echo true || echo false)"
+CURRENT_CONTEXT_NAME="$(kubectl config current-context view)"
+PLATFORM="self-managed"
+
+autoDetectEnvironment(){
+    if [[ -z "$CURRENT_CONTEXT_NAME" ]]; then
+        echo "no configuration has been provided"
+        return
+    fi
+
+    echo "Autodetecting environment"
+    if [[ $CURRENT_CONTEXT_NAME =~ ^minikube.* ]]; then
+        PLATFORM="minikube"
+        elif [[ $CURRENT_CONTEXT_NAME =~ ^gke_.* ]]; then
+        PLATFORM="gke"
+        elif [[ $CURRENT_CONTEXT_NAME =~ ^kind-.* ]]; then
+        PLATFORM="kind"
+        elif [[ $CURRENT_CONTEXT_NAME =~ ^k3d-.* ]]; then
+        PLATFORM="k3d"
+    fi
+}
+
+handleKubearmor(){
+	[[ "$1" == "" ]] && echo "no operation specified, specify apply/delete" && return 1
+    echo "$1 Kubearmor on $PLATFORM Kubernets Cluster"
+    case $PLATFORM in
+        gke)
+            kubectl $1 -f https://raw.githubusercontent.com/kubearmor/KubeArmor/master/deployments/GKE/kubearmor.yaml
+        ;;
+        microk8s)
+            microk8s kubectl $1 -f https://raw.githubusercontent.com/kubearmor/KubeArmor/master/deployments/microk8s/kubearmor.yaml
+        ;;
+        self-managed)
+            kubectl $1 -f https://raw.githubusercontent.com/kubearmor/KubeArmor/master/deployments/docker/kubearmor.yaml
+        ;;
+        containerd)
+            kubectl $1 -f https://raw.githubusercontent.com/kubearmor/KubeArmor/master/deployments/generic/kubearmor.yaml
+        ;;
+        minikube)
+            echo "Kubearmor cannot be installed on minikube. Skipping..."
+        ;;
+        kind)
+            echo "Kubearmor cannot be installed on kind. Skipping..."
+        ;;
+        *)
+            echo "Unrecognised platform: $PLATFORM"
+    esac
+}
+
+handleKnoxAutoPolicy()
+{
+	[[ "$1" == "" ]] && echo "no operation specified, specify apply/delete" && return 1
+	KNOXAUTOPOLICY_REPO="https://raw.githubusercontent.com/accuknox/knoxAutoPolicy-deployment/main/k8s"
+	KNOXAUTOPOLICY_SVC="$KNOXAUTOPOLICY_REPO/service.yaml --namespace explorer"
+	KNOXAUTOPOLICY_CFG="$KNOXAUTOPOLICY_REPO/dev-config.yaml --namespace explorer"
```

Comment:
I think this dev-config would not use explorer's mysql. As it is pointed to localhost.

Reply:
okk i see ... so essentially mysql db wont work then !

Reply:
I compared https://github.com/accuknox/explorer-charts/blob/main/autoPolicy/dev-config.yaml with https://github.com/accuknox/knoxAutoPolicy-deployment/blob/main/k8s/dev-config.yaml and the MySQL config seems same.

Reply:
I changed the mysql value, few minutes ago. And I think I should also change the hubble url to what it was before.